File: 1ºAno/Lista3/Python/Sala/files/ex4.py
while True:
    while True:
        vals = input("Na mesma linha, entre, respectivamente, com os valores referentes a:\n\n1) Número N\n2) Base da multiplicação sucessiva\n3) Índice do produto\nPara sair, entre com 0 em todos os valores\n\n").split()
        try:
            vals[0], vals[1], vals[2] = int(vals[0]), int(vals[1]), int(vals[2])
            break
        except:
            print ("*Número(s) inválido(s)... Tente novamente*")
    if vals[0] == vals[1] == vals[2] == 0:
        break
    i = 1
    while True:
        if (vals[0] * vals[1] ** i >= vals[2]):
            break
        print ("{} * {} ^ {} = {}" .format(vals[0], vals[1], i, vals[0] * vals[1] ** i))
        i += 1
    print ()
print ("\nAdeus!")

File: 1ºAno/Lista3/C/Sala/files/ex4.c
#include <stdio.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <math.h>

int main() {
    for (;;) {
        char *dados = (char *)malloc(100 * sizeof (char));
        int n = 0, mult = 0;
        unsigned long long stop = 0, result = 0;
        for (;;) {
            printf ("Na mesma linha, entre, respectivamente, com os valores referentes a:\n\n1) Valor de N\n2) Valor da multiplicação sucessiva\n3) Número de parada\n\n");
            fgets(dados, 100, stdin);
            char *pt = strtok(dados, " ");
            int i = 0;
            while (pt != NULL) {
                if (i == 0) {
                    n = atoi(pt);
                }
                else if (i == 1) {
                    mult = atoi(pt);
                }
                else {
                    stop = atoi(pt);
                }
                pt = strtok(NULL, " ");
                ++i;
            }
            if (stop == 0 && mult != n || n == 0) {
                printf ("*Número(s) inválido(s)... Tente novamente*\n");
            }
            else {
                break;
            }
        }
        if (n == 0 && mult == 0 && stop == 0) {
            break;
        }
        for (int i = 1;; ++i) {
            result = n * pow(mult, i);
            if (result >= stop) {
                break;
            }
            printf ("%d * %d ^ %d = %llu\n", n, mult, i, result);
        }
    }
    return 0;
}